openlibrary/macros/databarView.html

```diff
@@ -1,7 +1,10 @@
-$def with (page)
+$def with (page, edition=None)
 
 $if page.type.key in ["/type/work", "/type/edition", "/type/author"]:
-    $ edit_url = page.url(suffix="/edit")
+    $if page.type.key in ["/type/work"]:
+        $ edit_url = edition.key + "/" + edition.title + "/edit"
```

Comment:
This will create invalid urls if edition.title has special chars; use `edition.url(suffix="/edit")` if it supports it. Also only if edition is defined.